nabs/_html.py

```diff
@@ -0,0 +1,108 @@
+"""
+html templates for use in elog post formatting
+"""
+
+collapse_list_head = """<html>
+<head>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<style>
+.collapsible {
+  background-color: #777;
+  color: white;
+  padding: 10px;
+  width: 100%;
+  border: none;
+  text-align: left;
+  outline: none;
+  font-size: 15px;
+}
+
+.collapsible:hover {
+  background-color: #555;
+}
+
+.collapsible:after {
+  content: '+';
+  color: white;
+  font-weight: bold;
+  float: right;
+  margin-left: 5px;
+}
+
+.active:after {
+  content: "-";
+}
+
+.content {
+  min-width: 100px;
+  overflow-x: auto;
+}
+
+.parent,
+.child {
+  display: none;
+  margin-left: 20px;
+}
+
+.parent.show,
+.child.show {
+  display: block;
+}
+</style>
+</head>
+<body>
+"""
+
+collapse_list_tail = """<script>
+// manage +/- toggle
+var coll = document.getElementsByClassName("collapsible");
+var i;
+
+for (i = 0; i < coll.length; i++) {
+  coll[i].addEventListener("click", function() {
+    this.classList.toggle("active");
+    var content = this.nextElementSibling;
+  });
+}
+
+// manage show/hide behavior
+var allCollapsibles = document.querySelectorAll('.collapsible');
+allCollapsibles.forEach( item => {
+  item.addEventListener("click", function() {
+     if(this.nextElementSibling.classList.contains('show')) {
+       this.nextElementSibling.classList.remove('show')
+     } else {
+       this.nextElementSibling.classList.add('show')
+     }
+  });
+});
+</script>
+
+</body>
+</html>"""
+
+run_table_head = """<!DOCTYPE html>
+<html>
```

Comment:
Preface - I'm no html expert by any stretch.

But there's something a bit weird about embedding `<html>` tags _inside_ of an existing html document. (*)
When you view the source of the elog page with these embedded, how is it structured? Do you see `<html> .... <html> ophyd obj info </html> ... </html>`

If each post is embedded in an `iframe` that's a different story (but I'd guess it's not the case).

I think the "proper" solution would be either getting Murali to modify elog pages to add our css classes for these tables, or on a per-element basis set `style=` (which is a bit messy for sure).

I think this probably shouldn't hold up this PR, but we should look into the right way of doing things to ensure future compatibility with browsers viewing these log entries.

(*) https://www.w3schools.com/html/html_css.asp these are the three "acceptable" ways of embedding style

Reply:
Good points, it seems the elog strips my html and body tags, so I guess there's some parsing going on at the elog end.  

I tried to keep the css/js as minimal as possible, but if we start to post more things with more involved css I can see asking Murali to hold on to our css classes.